bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/i18n/AbstractI18nDiscoveryService.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.config.discovery.i18n;
+
+import java.util.Locale;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.internal.DiscoveryResultImpl;
+import org.eclipse.smarthome.core.i18n.I18nUtil;
+import org.eclipse.smarthome.core.i18n.TranslationProvider;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.FrameworkUtil;
+
+/**
+ * The {@link AbstractI18nDiscoveryService} extends the {@link AbstractDiscoveryService} to allow localizing
+ * a {@link DiscoveryResult}of a thing discovery using the I18N mechanism of the Eclipse SmartHome framework.
+ * Currently the label of the {@link DiscoveryResult} is the single attribute which can be localized.
+ * It automatically infers the key if the default text is not a constant.
+ *
+ * @author Laurent Garnier - Initial contribution
+ */
+public abstract class AbstractI18nDiscoveryService extends AbstractDiscoveryService {
+
+    protected TranslationProvider i18nProvider;
+    protected Locale locale;
+
+    /**
+     * Creates a new instance of this class with the specified parameters.
+     *
+     * @param supportedThingTypes
```

Comment:
@kaikreuzer Can we add a guideline for the JavaDoc format in the source code?
I don't see any reason why we should add a line break between the parameter name and the description.
I don't see any reason to add empty lines between `@param`, `@return`, `@throws` etc.

Reply:
http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html

`@param`

```
Additional spaces can be inserted between the name and description so that the descriptions line up in a block.
```

The only about new lines in general is between description and tags.

```
Insert a blank comment line between the description and the list of tags, as shown.
```

I know that there is no "forbidden" stated in that document but at least we in our project should use the same format (at least for every newly created one). So can we agree to drop that line breaks after the parameter name (exception name) and also the empty lines between the tags?

Reply:
For information, I just copied/pasted this from the parent (existing) class without any change.

Reply:
@lolodomo Thanks for that information.

I know that there are some usage of that style. I would not create a comment that you need to change an existing one that is not related to your changes.
I just want to ensure that new code / documentation is using the style we use.

As long as there is no guideline -- so "the style we have to use" -- I just could ask you to change it but I would like to know if @kaikreuzer and @sjka agree with me.

Reply:
But maybe the correct approach in my case would be to just inherit the java doc rather than duplicating it ?

Reply:
> but I would like to know if @kaikreuzer and @sjka agree with me.

Yes, I agree. But as this is code formatting, I would refrain from specifically adding it to the guidelines (because we have a million other such things like where to put the brackets...) - we simply should make sure that the existing code base follows that scheme.